chip8_vm/src/instructions.rs

```diff
@@ -1,11 +1,45 @@
 //! Raw and high-level instruction abstractions
 
-/// First register in an opcode
+use std::num::{FromPrimitive, ToPrimitive};
+
+/// A register index/name
 ///
 /// There are 16 registers, `V0`..`VF`
-pub type Vx = u8;
+#[derive(FromPrimitive, Copy, Debug)]
```

Comment:
FYI: `FromPrimitive` is marked as unstable:

```
src/instructions.rs:8:10: 8:23 warning: use of unstable library feature 'core': trait is likely to be removed
src/instructions.rs:8 #[derive(FromPrimitive, Copy, Debug)]
```

We are currently ignoring such warnings with `#![feature(core)]` Not super worried about it just nice to be aware. I'm not sure what the planned replacement is if it is being removed.